src/main/scala/xiangshan/mem/lsqueue/StoreMisalignBuffer.scala

```diff
@@ -195,36 +241,60 @@ class StoreMisalignBuffer(implicit p: Parameters) extends XSModule
           bufferState := s_wb
           globalException := hasException
           globalMMIO := isMMIO
-        } .elsewhen(io.splitStoreResp.bits.need_rep || (unSentStores & ~clearOh).orR) {
+        } .elsewhen(io.splitStoreResp.bits.need_rep || (unSentStores & (~clearOh).asUInt).orR) {
           // need replay or still has unsent requests
           bufferState := s_req
         } .otherwise {
           // got result, goto calculate data and control sq
-          bufferState := s_cal
+          bufferState := s_wb
         }
       }
     }
 
-    is (s_cal) {
-      when (io.sqControl.storeInfo.dataReady) {
-        bufferState := s_sq_req
+    is (s_wb) {
+      when (req.isvec) {
+        when (io.vecWriteBack.map(x => x.fire).reduce( _ || _)) {
+          bufferState := s_idle
+          req_valid := false.B
+          curPtr := 0.U
+          unSentStores := 0.U
+          unWriteStores := 0.U
+          globalException := false.B
+          globalMMIO := false.B
+          isCrossPage := false.B
+          needFlushPipe := false.B
+        }
+      }
+      when (io.writeBack.fire && (!isCrossPage || globalMMIO || globalException)) {
```

Comment:
It feels like the juxtaposition condition shouldn't be used here `when`, but rather `otherwise`

Reply:
Sorry, I don't understand what you mean.
When `isCrossPage`, `io.writeBack.fire` does not mean that `StoreMisalignBuffer` can be cleared. We also need to wait for `Rob`. And if this cross-page `store` triggers an exception or `MMIO`, then it can be cleared directly when writing back.